src/rpm/headers/header.rs

```diff
@@ -308,7 +308,7 @@ impl Header<IndexSignatureTag> {
     /// PGP and RSA tags expect signatures according to [RFC2440](https://tools.ietf.org/html/rfc2440)
     ///
     /// Please use the [`builder`](Self::builder()) which has modular and safe API.
-    #[cfg(feature = "signature-meta")]
+    #[allow(unused)]
     pub(crate) fn new_signature_header(
```

Comment:
Isn't this decoupling the signing from the signature creation?  I'm not 100% sure I understand what you mean.